### DISCUSSION TITLE
Dremel 3D40 port and bootloader

### DISCUSSION THREAD

**Discussion by kevinallein** [moonglow/FlashForge_Marlin#50] [Ideas]
Hi all, that might be a bit of a long shot. I have a Dremel 3D40, that had a defective display and since that display is actually custom to Bosch (the manufacturer will not sell to anybody else), I thought this is the chance to convert the printer to Marlin. I did not find anything on 3D40, but only this port here.
The 3D40 uses an STM32F103RCT6 like several other printer boards and in addition a TI Stellaris as a plugin module for GUI and higher level control.
So I thought of running Marlin on the STM and completely remove the TI Stellaris and connect a Raspberry PI with Octoprint through USB.
Obviously this means several things:
- get a USB bootloader onto the STM to allow downloading Marlin
- configure Marlin for this board. It uses TMC5130 motor drivers, the configuration.h in this project seems to use A4988 - is this really true ?

Anybody else interested in this ?

**Comment by moonglow** [comment]
I think if you provide us more information such as motherboard photo and so on it be useful. Another thing about Marlin porting process it is not just firmware configuration file i think it can be tricky, first of all you need to reverse engineering motherboard schematic to get all pins map and used peripheral devices. If we can compare to other FF printers ( not listed here ) they can use powerful MCU for GUI and G-code processor and stm32 as realtime stepper. And ofcause if someone will helps with this they need to have actual hardware on hands.

**Comment by kevinallein** [comment]
![Dremel_MB](https://user-images.githubusercontent.com/87988443/129154129-3e929bfa-6226-4651-9f2a-1d8f0eb1c3ae.jpg)

Here is a picture of the motherboard with all the metal covers removed. I have started to do some reengineering, althouth it is quite difficult on this PCB. The TI Sitara is on a plugin module, since I do not think anyone would have a firmware for that I was just thinking to remove that. I fear some soldering will be required to e.g. connect the USB port to the STM32, so I guess that might be a blocker for many people.

**Reply by moonglow** [reply]
Thanx ! TI Sitara based module with Linux. I think it much more easy to figure-out TFT display module pinout and find something that will be capable with old one. I think this is used with regular parallel TFT interface without internal controller.

**Comment by kevinallein** [comment]
The question is, do they actually run Linux on the Sitara ? If so, Dremel would actually have to make the Linux kernel available somewhere, but I cannot find anything. Since I cannot configure  my printer due to the display problems, could someone please check, whether you could actually telnet into the Sitara ?

**Comment by kevinallein** [comment]
Are all FF printers actually using closed source firmware or can you find any info about what they are using ?

**Reply by moonglow** [reply]
Yep, they sourcecode and schematic are closed, but for non linux based system they use they own highly moded version sailfish firmware they ported it to 32bit IC and also looks like they use Segger GUI library emWin for UI.  Check this [link](https://github.com/pressreset/CrashForge) maybe it will helps you to find a way how to get access to printer linux console. Dremel use FF guys as a developers for they printers.
